File: PersonalBlogCMS/api/utils/response.ts
import type { Response } from 'express';
import type { ApiResponse } from '../../shared/types.js';

export function success<T>(res: Response, data?: T, message = 'success'): Response<ApiResponse<T>> {
  return res.json({
    code: 0,
    message,
    data,
  });
}

export function error(res: Response, message: string, code = 1, status = 400): Response<ApiResponse> {
  return res.status(status).json({
    code,
    message,
  });
}

export function unauthorized(res: Response, message = '未授权访问'): Response<ApiResponse> {
  return error(res, message, 401, 401);
}

export function forbidden(res: Response, message = '禁止访问'): Response<ApiResponse> {
  return error(res, message, 403, 403);
}

export function notFound(res: Response, message = '资源不存在'): Response<ApiResponse> {
  return error(res, message, 404, 404);
}

File: OnlineRepairBooking/backend/pkg/response/response.go
package response

import (
	"net/http"

	"github.com/labstack/echo/v4"
)

type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

func Success(c echo.Context, data interface{}) error {
	return c.JSON(http.StatusOK, Response{
		Code:    0,
		Message: "success",
		Data:    data,
	})
}

func SuccessWithMessage(c echo.Context, message string, data interface{}) error {
	return c.JSON(http.StatusOK, Response{
		Code:    0,
		Message: message,
		Data:    data,
	})
}

func Error(c echo.Context, code int, message string) error {
	return c.JSON(http.StatusOK, Response{
		Code:    code,
		Message: message,
	})
}

func ErrorWithData(c echo.Context, code int, message string, data interface{}) error {
	return c.JSON(http.StatusOK, Response{
		Code:    code,
		Message: message,
		Data:    data,
	})
}

func BadRequest(c echo.Context, message string) error {
	return Error(c, http.StatusBadRequest, message)
}

func Unauthorized(c echo.Context, message string) error {
	return Error(c, http.StatusUnauthorized, message)
}

func Forbidden(c echo.Context, message string) error {
	return Error(c, http.StatusForbidden, message)
}

func NotFound(c echo.Context, message string) error {
	return Error(c, http.StatusNotFound, message)
}

func InternalServerError(c echo.Context, message string) error {
	return Error(c, http.StatusInternalServerError, message)
}
